crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,217 @@
+
+impl Cmd {
+    pub async fn run(&mut self) -> Result<(), Error> {
+        let (tx, mut rx) = mpsc::channel::<Message>(100);
+        let rebuild_state = Arc::new(Mutex::new(false));
+        let workspace_root: &Path = self
+            .build_cmd
+            .manifest_path
+            .parent()
+            .unwrap_or_else(|| Path::new("."));
+        let env_toml_path = Arc::new(workspace_root.join("environments.toml"));
+        let env_toml_parent = Arc::new(
+            env_toml_path
+                .parent()
+                .unwrap_or(Path::new(""))
+                .to_path_buf(),
+        );
+
+        let mut watched_dirs = Vec::new();
+        let packages = self
+            .build_cmd
+            .list_packages()?
+            .into_iter()
+            .map(|package| {
+                Arc::new(PathBuf::from(
+                    package.manifest_path.parent().unwrap().as_str(),
+                ))
+            })
+            .collect::<Vec<_>>();
+
+        for package_path in &packages {
+            watched_dirs.push(package_path.clone());
+            eprintln!("Watching {}", package_path.as_path().display());
+        }
+        let watched_dirs_clone = watched_dirs.clone();
+        let env_toml_path_clone = Arc::clone(&env_toml_path);
+        let env_toml_parent_clone = Arc::clone(&env_toml_parent);
+        let mut watcher =
+            notify::recommended_watcher(move |res: Result<notify::Event, notify::Error>| {
+                if let Ok(event) = res {
+                    if matches!(
+                        event.kind,
+                        notify::EventKind::Create(_)
+                            | notify::EventKind::Modify(_)
+                            | notify::EventKind::Remove(_)
+                    ) {
+                        if let Some(path) = event.paths.first() {
+                            // Ignore temporary files
+                            if is_temporary_file(path) {
+                                return;
+                            }
+                            let env_toml_parent_abs = canonicalize_path(&env_toml_parent_clone);
+                            let env_toml_path_abs = canonicalize_path(&env_toml_path_clone);
+                            let parent_is_env_toml_parent =
+                                path.parent() == Some(env_toml_parent_abs.as_path());
+                            let path_is_env_toml = path == env_toml_path_abs.as_path();
+                            let parent_is_in_watched_dirs = is_parent_in_watched_dirs(
+                                &env_toml_parent_abs,
+                                &watched_dirs_clone,
+                            );
+                            // skip if the file is in the parent directory of environments.toml and it is not environments.toml
+                            if !(parent_is_env_toml_parent
+                                && !path_is_env_toml
+                                && !parent_is_in_watched_dirs)
```

Comment:
```suggestion
                            if !parent_is_env_toml_parent
                                || path_is_env_toml
                                || parent_is_in_watched_dirs
```
It's best to distribute the negation as it's less work for the reader.
So in this case if `parent_is_in_watched_dirs` is true which is independent of the `path` argument this is invoked?

Reply:
yes because any file in the watched dirs should trigger a rebuild.

Reply:
basically we have to add the parent dir of the env file to the list of dirs in order to check if there is a deletion or creation of the env file. If that dir is already in the watched dirs then all watched files should trigger a rebuild

Reply:
I realized the parent_is_env_toml_parent check is redundant so its been removed in the refactored version